functional/tdx/run.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+script_path=$(dirname "$0")
+source "${script_path}/../../lib/common.bash"
+
+runtime_type="io.containerd.kata.v2"
+config_file=""
+kernel_tdx_path="/usr/share/kata-containers/vmlinuz-tdx.container"
+qemu_tdx_path="/usr/local/bin/qemu-system-x86_64"
+container_name=test
+FIRMWARE="${FIRMWARE:-}"
+FIRMWARE_VOLUME="${FIRMWARE_VOLUME:-}"
+
+trap cleanup EXIT
+
+get_config_file() {
+	if [ -z "${config_file}" ]; then
+		for f in $(kata-runtime --show-default-config-paths); do
+			[ -f "${f}" ] && config_file="${f}" && break
+		done
+	fi
+	echo "${config_file}"
+}
+
+cleanup() {
+	local config_file="$(get_config_file)"
+	sudo mv -f "${config_file}.bak" "${config_file}"
+
+	sudo ctr t kill -s 15 ${container_name} || true
+	sudo ctr t rm -f ${container_name} || true
+	sudo ctr c rm ${container_name} || true
+
+	clean_env_ctr
+}
+
+enable_confidential_computing() {
+	local conf_file="$(get_config_file)"
+	[ -n "${conf_file}" ] || die "configuration file not found"
+	sudo sed -i -e 's|path =.*|path = "'${qemu_tdx_path}'"|' \
+		-e 's|^kernel =.*|kernel = "'${kernel_tdx_path}'"|' \
+		-e 's|^# confidential_guest.*|confidential_guest = true|' \
+		-e 's|^kernel_params =.*|kernel_params = "force_tdx_guest tdx_disable_filter"|' \
+		-e 's|^firmware =.*|firmware = "'${FIRMWARE}'"|' \
+		-e 's|^firmware_volume =.*|firmware_volume = "'${FIRMWARE_VOLUME}'"|' \
+		-e 's|^cpu_features.*|cpu_features = "pmu=off,-kvm-steal-time"|' \
+		${conf_file}
+}
+
+install_kernel_tdx() {
+	local kernel_url="http://jenkins.katacontainers.io/job/kata-containers-2.0-kernel-tdx-x86_64-nightly/lastSuccessfulBuild/artifact/artifacts"
+	local latest=$(curl ${kernel_url}/latest)
+	curl -L ${kernel_url}/vmlinuz-${latest} -o vmlinuz-tdx.container
+	sudo mv -f vmlinuz-tdx.container ${kernel_tdx_path}
+}
+
+install_qemu_tdx() {
+	local qemu_url="http://jenkins.katacontainers.io/job/kata-containers-2.0-qemu-tdx-x86_64/lastSuccessfulBuild/artifact/artifacts/kata-static-qemu.tar.gz"
+	curl "${qemu_url}" | sudo tar --strip-components=1 -C /usr/local/ -zxf -
+}
+
+run_test() {
```

Comment:
I saw somewhere a demo for confidential containers where it was executed a QMP command to dump the memory of the VM to show it was really encrypted. @devimc if you are interested on adding that checking as well, I can have a look on how to exactly run the qmp command.

Reply:
@wainersm yes please, this would be *very very* useful

Reply:
Hi @devimc , absolutely, it can be done in a follow up PR.

Reply:
@devimc just found it, it is in a demo carried out by IBM. See https://youtu.be/1v1hxAWwHDo?t=389 (6'29").

Basically he exported an environment variable within the container, then ran `dump-guest-memory' (an QMP command) to dump the memory in a file and finally grep'ed the content (memory is encrypted so he could not find the word).